sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/models/AzureJsonHelper.java

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.monitor.opentelemetry.exporter.implementation.models;
+
+import com.azure.json.JsonWriter;
+import com.azure.json.WriteValueCallback;
+
+import java.io.IOException;
+import java.util.Map;
+
+public final class AzureJsonHelper {
+
+    // this method is replacing JsonWriter.writeMapField(), which would generate json attributes with null value.
+    // This method is used to write a map to a json writer.
+    // The map is written as a json object with the key as the field name and the value as the field value.
+    // The value is written using the writeValueCallback.
+    public static <T> void writeMap(Map<String, T> map, String fieldName, JsonWriter jsonWriter, WriteValueCallback<JsonWriter, T> function) throws IOException {
+        if (map == null) {
+            return;
+        }
+        jsonWriter.writeFieldName(fieldName);
+        jsonWriter.writeStartObject();
+        for (Map.Entry<String, T> entry : map.entrySet()) {
+            T value = entry.getValue();
+            if (value == null) {
+                continue;
+            }
```

Comment:
can you remind me which attributes in the `properties` map had null values?

Reply:
tags, like "ai.cloud.role".